freedb/protocols.md

```diff
@@ -1,8 +1,311 @@
+with the range where the data lives in [A1 notation format][A1Notation]. It's recommended to select the entire column
+for the `<data_range>`, e.g. suppose we have 5 columns (including `_rid` metadata column) in `Sheet1` we should
+use `Sheet1!A:E` to select the entire rows.
+
+To construct the value of `values` parameter (inside the request's body), convert each row into an array where the
+1st value (value of the `_rid` metadata column) equals to `"=ROW()"` and `i`-th (1-based index) value contains the
+value of `i`-th column of the current row.
+
+Suppose we want to insert the data below:
+```
+[
+    Person(name="A", age=1), # 1st row
+    Person(name="B", age=2), # 2nd row
+    Person(name="C", age=3), # 3rd row
+    Person(name="D", age=4)  # 4th row
+]
+```
+
+then the `values` would look like this:
+```
+"values": [
+    ["=ROW()", "A", 1],
+    ["=ROW()", "B", 2],
+    ["=ROW()", "C", 3],
+    ["=ROW()", "D", 4],
+]
+```
+
+#### Select
+
+Returns the number of rows that matched the given condition.
+
+The operation method should expect these parameters:
+| Parameter Name | Required | Remarks                                                                                                                                                        |
+| -------------- | -------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `columns`      | No       | List of columns that we will fetch. Defaults to return all columns if not provided.                                                                            |
+| `conditions`   | No       | Condition that must be true for the row to be returned in [Google Sheet Query Language][QueryLanguage] format. Defaults to return all columns if not provided. |
+| `limit`        | No       | Impose a limit on the number of rows that will be returned. Defaults to no limit if not provided.                                                              |
+| `offset`       | No       | Skips the given number of first rows from being returned. Defaults to not skip any rows if not provided.                                                       |
+| `order_by`     | No       | Sorts the row based on the given column values in either ascending or descending order. Defaults to not sort the rows.                                         |
+
+We will utilise the [GViz API][GVizAPI] to return the matching rows from the spreadsheet by running the Google Query below:
+
+`SELECT <column-1>, <column-2>, ..., <column-n> WHERE A IS NOT NULL [AND <condition>] [ORDER BY <ordering-1>, <ordering-2>, ..., <ordering-n>] [LIMIT <limit>] [OFFSET <offset>]`
+
+Note: Query inside square bracket denotes an optional query, only need to be provided if necessary.
+
+- Replace the `<column-1>, <column-2>, ..., <column-n>` based on the columns that are specified in the `columns` parameter.
```

Comment:
I don't think this is accurate, right? The `columns` parameter is still using the application level column name, not the A, B, C model, right?

Reply:
Removed this part (related to this https://github.com/FreeLeh/docs/pull/5#discussion_r967126799)